### PR TITLE
Map displayed using OL3 [WIP]

### PR DESCRIPTION
**Pull request by pgiraud**
Note: work is still in progress

With this pull request, the map is rendered using OpenLayers3 which allows us to use any tile service as base map.
See #34 and #41.
For now, the chosen basemap is `Stamen Watercolor` because it's not intrusive (no labels and pastel colors). It can be changed eventually.

The overlay layer showing the administrative divisions and hazard choropleths is the result of a request to the mapnik service (with almost no change). This image is computed so that it covers the administrative area but not more.

What's missing for now is the interaction (drill down).

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -22,6 +22,7 @@
     "jquery": "~2.1.3",
     "jshint": "~2.5.1",
     "less": "~1.7.5",
+    "openlayers": "~3.3.0",
```
Comment (elemoine) [inline package.json]: We can use 3.5.0 now.

File: thinkhazard/templates/report.jinja2 [hunk 2]
```diff
@@ -126,3 +125,14 @@
...
 {% endblock %}
+
+{% block scripts %}
+{% if debug %}
+    <script src="{{('%s/openlayers/dist/ol-debug.js' % node_modules)|static_url}}"></script>
+{% else %}
+    <script src="{{('%s/openlayers/dist/ol.js' % node_modules)|static_url}}"></script>
+{% endif %}
+    <script>
+      var division_bounds = {{bounds}};
```
Comment (elemoine) [inline thinkhazard/templates/report.jinja2]: Please use the camelCase notation for js variables.

File: thinkhazard/views/report.py [hunk 3]
```diff
@@ -68,5 +70,13 @@ def report(request):
     hazard_data = hazard_data.values()
     hazard_data = sorted(hazard_data, key=lambda d: d['categorytype'].order)
 
+    # Get the geometry for division and compute its extent
+    # FIXME use box2d
+    division_geometry = DBSession.query(AdministrativeDivision.geom) \
+        .filter(AdministrativeDivision.code == division_code).one()
```
Comment (elemoine) [inline thinkhazard/views/report.py]: We already retrieved the administrative division with Query 1, no?

File: thinkhazard/templates/base.jinja2 [hunk 4]
```diff
@@ -31,7 +31,7 @@
       var app = {};
       app.administrativedivisionUrl = '{{'administrativedivision'|route_url}}';
       app.reportpageUrl = '{{'report'|route_url}}';
-      app.mapImgUrl = '{{'map'|route_url(format='jpeg')}}';
+      app.mapImgUrl = '{{'map'|route_url(format='png')}}';
```
Comment (elemoine) [inline thinkhazard/templates/base.jinja2]: @fvanderbiest was the one suggesting to use jpeg. Why did you change it?

File: thinkhazard/static/js/report.js [hunk 5]
```diff
@@ -210,7 +97,40 @@
...
+    showHazard();
     $('#legend').hide();
   }
+
+  /**
+   * Adds or replaces the hazard image layer.
+   */
+  function showHazard(hazardType) {
+    if (hazardLayer) {
+      map.removeLayer(hazardLayer);
```
Comment (elemoine) [inline thinkhazard/static/js/report.js]: There's a `setSource` method that we could use instead of re-creating the layer.

Comment (ingenieroariel) [inline thinkhazard/static/js/report.js]: Can we use this basemap instead of stame watercolor?

https://a.tiles.mapbox.com/v4/ingenieroariel.m9a2h374/page.html?access_token=pk.eyJ1IjoiaW5nZW5pZXJvYXJpZWwiLCJhIjoibXhDZ3pIMCJ9.qTmPYCbnUKtaNFkvKKysAQ#3/0.00/0.00

File: thinkhazard/templates/base.jinja2 (hunk 4 above)
Comment (pgiraud) [inline thinkhazard/templates/base.jinja2]: This is because I want the overlay layer to be transparent.

**Comment by pgiraud** [comment]
Obsolete pull request. See #143.
